### PR TITLE
[sk] Updated directory tree generation

### PR DESCRIPTION
**Pull request by skunichetty**
# Summary
Tree generation for frontend filesystem. Example payload: 
image

# Tests
Tests were performed locally to make sure that the correct tree structure is generated

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/models/file.py [hunk 1]
```diff
@@ -54,10 +43,45 @@ def delete(self):
...
+    disabled = name[0] == '.' or name in INACCESSIBLE_DIRS
+    children = []
+    entries = os.scandir(directory)
+    children.extend(
+        _traverse(entry, disabled, 1) for entry in sorted(entries, key=lambda entry: entry.name)
+    )
+    return dict(name=name, children=children)
+
+
+def _traverse(entry, disabled=True, depth=1) -> Dict:
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/file.py:72]: How are you using depth? Do you want to stop traverse when depth > MAX_DEPTH?

File: mage_ai/data_preparation/models/file.py [hunk 2]
```diff
@@ -54,10 +43,45 @@ def delete(self):
...
+    Traverses directory tree and returns the underlying file structure
+
+    Args:
+        directory (str): Top-level directory to generate tree from
+
+    Returns:
+        Dict: Directory tree payload representation
+    """
+    name = os.path.splitext(os.path.basename(directory))[0]
+    disabled = name[0] == '.' or name in INACCESSIBLE_DIRS
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/file.py:63]: There are some duplicate code here.
You can define `_traverse` method as
```
def _traverse(name, path, disabled=True, depth=1) -> Dict:
    ....
```
Then call `_traverse(repo_name, repo_path, False, 0)`

Comment (skunichetty) [reply mage_ai/data_preparation/models/file.py]: Fixed, although function has a lot of parameters now

File: mage_ai/data_preparation/models/file.py (hunk 1 above)
Comment (skunichetty) [reply mage_ai/data_preparation/models/file.py]: A check against max depth accidentally got deleted at some point, I added it back. I stop traversal when `depth > MAX_DEPTH`.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**